presto-verifier/src/main/java/com/facebook/presto/verifier/framework/VerificationManager.java

```diff
@@ -147,6 +162,33 @@ public void close()
                 }
             }
         }
+        executor.shutdownNow();
+    }
+
+    @Override
+    public synchronized boolean resubmit(Verification verification, QueryException queryException)
+    {
+        if (!shouldRequeue(queryException)) {
+            return false;
+        }
+
+        VerificationRef ref = new VerificationRef(verification);
+        Integer requeudCount = requeuedCounts.getOrDefault(ref, 0);
+        if (requeudCount < failureRequeueLimit) {
```

Comment:
nit: my personal preference - 

```
if (requeuedCount >= failureRequeueLimit) {
  return false;
}

.... rest of the success case logic
```

This just makes it easier to add other dropping conditions.